AnnService/src/IndexSearcher/main.cpp

```diff
@@ -237,37 +254,76 @@ int Process(std::shared_ptr<SearcherOptions> options, VectorIndex& index)
             totalLatency[mc] += batchLatency;
         }
 
-        if (fp.is_open())
+        if (fp != nullptr)
         {
-            fp << std::setprecision(3) << std::fixed;
-            for (SizeType i = 0; i < numQuerys; i++)
-            {
-                if (queryMetas != nullptr) {
-                    ByteArray qmeta = queryMetas->GetMetadata(startQuery + i);
-                    fp.write((const char*)qmeta.Data(), qmeta.Length());
-                }
-                else {
-                    fp << i;
-                }
-                fp << ":";
-                for (int j = 0; j < options->m_K; j++)
+            if (options->m_outputformat == 0) {
+                for (SizeType i = 0; i < numQuerys; i++)
                 {
-                    if (results[i].GetResult(j)->VID < 0) {
-                        fp << results[i].GetResult(j)->Dist << "@NULL" << std::endl;
-                        continue;
-                    }
-
-                    if (!options->m_withMeta) {
-                        fp << (results[i].GetResult(j)->Dist / baseSquare) << "@" << results[i].GetResult(j)->VID << std::endl;
+                    if (queryMetas != nullptr) {
```

Comment:
If we want to support multiple output formats, should we wrap it in a `VectorSetWriter` the way we do with multiple input formats? I think we have this kind of logic in SSDServing too

Reply:
Yes, good suggestion! We can do it after this pull request.